bluez-async/src/lib.rs

```diff
@@ -301,22 +301,26 @@ impl BluetoothSession {
     }
 
     /// Get a list of all Bluetooth adapters on the system.
-    async fn get_adapters(&self) -> Result<Vec<AdapterId>, dbus::Error> {
+    async fn get_adapters(&self) -> Result<Vec<AdapterId>, BluetoothError> {
         let bluez_root = Proxy::new(
             "org.bluez",
-            "/",
+            "/org/bluez",
             DBUS_METHOD_CALL_TIMEOUT,
             self.connection.clone(),
         );
-        // TODO: See whether there is a way to do this with introspection instead, rather than
-        // getting lots of objects we don't care about.
-        let tree = bluez_root.get_managed_objects().await?;
-        Ok(tree
-            .into_iter()
-            .filter_map(|(object_path, interfaces)| {
-                interfaces
-                    .get(ORG_BLUEZ_ADAPTER1_NAME)
-                    .map(|_| AdapterId { object_path })
+        let root_node = bluez_root.introspect_parse().await?;
```

Comment:
It can fail either because the D-Bus call failed, or because we failed to parse the XML. We return an appropriate error in either case. What did you have in mind, a log message?

Reply:
I was thinking of wrapping the error in something that suggests that bluez might not be running, and that you might want to start it or something. Maybe the underlying dbus-rs error is helpful enough in this case? I haven't checked.